src/frontend/src/optimizer/plan_node/logical_changelog.rs

```diff
@@ -141,15 +140,29 @@ impl ToBatch for LogicalChangeLog {
 
 impl ToStream for LogicalChangeLog {
     fn to_stream(&self, ctx: &mut ToStreamContext) -> Result<StreamPlanRef> {
-        let new_input = self.input().to_stream(ctx)?;
-
+        let input = self.input().to_stream(ctx)?;
+        let dist = input.distribution();
+        let (distribution_keys, new_input) = match dist {
+            Distribution::HashShard(distribution_keys)
+            | Distribution::UpstreamHashShard(distribution_keys, _) => {
+                (distribution_keys.clone(), input)
+            }
+            _ => {
+                let stream_key = input
+                    .stream_key()
+                    .ok_or_else(|| anyhow::anyhow!("changelog input must have a stream key"))?
+                    .to_vec();
```

Comment:
I guess we can assume the stream key must exist for stream plan node? Then we can call `expect_stream_key`. cc @chenzl25